python_modules/dagster/dagster/_core/host_representation/external_data.py

```diff
@@ -785,12 +787,14 @@ def __new__(
         upstream_asset_key: AssetKey,
         input_name: Optional[str] = None,
         output_name: Optional[str] = None,
+        partition_mapping: Optional[PartitionMapping] = None,
     ):
         return super(ExternalAssetDependency, cls).__new__(
             cls,
             upstream_asset_key=upstream_asset_key,
             input_name=input_name,
             output_name=output_name,
+            partition_mapping=partition_mapping,
```

Comment:
should we have a check that the provided partition mapping is serializable / one of the partition mappings we've whitelisted, and if not, raise an error?

this is a breaking change for some people, I'd imagine. the alternative would be to silently ignore the partition mapping if it wasn't whitelisted, or put in a placeholder like "unserializable partition mapping"

Reply:
Great point. I added some handling for this. Now we consider non-builtin partition mappings as deprecated and we warn on them at asset construction time.  If you do use a non-builtin partition mapping, we revert to the default in host processes.